supabase/functions/auth/index.ts

```diff
@@ -0,0 +1,151 @@
+import { serve } from "https://deno.land/std/http/server.ts";
+import { createClient } from "https://esm.sh/@supabase/supabase-js";
+import { db } from "../_shared/db.ts";
+import { students } from "../_shared/schema.ts";
+import { getCookies, setCookie } from "https://deno.land/std/http/cookie.ts";
+
+type ValidationSchema = {
+  [key: string]: {
+    required?: boolean;
+    min?: number;
+    enum?: string[];
+  };
+};
+
+//need to move out the client create
+const supabase = createClient(
+  Deno.env.get("SUPABASE_URL")!,
+  Deno.env.get("SUPABASE_ANON_KEY")!,
+);
+
+const validate = (
```

Comment:
Re-write this using `Zod` and safeParse() the function. It's a cleaner solution.